### DISCUSSION TITLE
How to share Ceph pools between multiple storage types?

### DISCUSSION THREAD

**Discussion by mmeier86** [rook/rook#12841] [Q&A] [answered]
Hello,

I'm planning to replace my current setup with a Kubernetes cluster, and I'm considering moving my baremetal Ceph cluster to Rook. Reading through the docs, I have got one question: Is it possible to use one pool for several storage types with Ceph Rook?

For example, in my current setup I have one pool with HDDs and one with SSDs. The HDDs pool is used as the backing storage for my CephFS, some of my RGW storage classes as well as some of my RBDs. But from Rook's docs, it looks like every storage type CRD contains the definition for a new pool to be created for that specific usage, instead of being able to just supply a pool name of an existing pool?

**Comment by travisn** [comment, marked as answer]
@mmeier86 The pool created by the CephBlockPool will have any desired name. If the pool already exists, Rook will just start managing it. But the pools created by CephObjectStore and CephFilesystem have separate pools created specifically for them and Rook doesn't provide a way to share the pools. 

What is the high level goal? If you're trying to have certain pools select certain OSDs based on type (ssd/hdd), you can set the `deviceClass` property on the pools to select that class of OSDs.

**Reply by mmeier86** [reply]
Thanks for the answer! My main goal was really just to keep the PGs per OSD reasonable as the Ceph docs advise, as my cluster is going to be rather small with 6 OSDs total for the foreseeable future.
